### PR TITLE
Windows Github Actions now uses MSYS2

### PR DESCRIPTION
**Pull request by Jasagredo**
# Description

Windows Github actions CI makes use of chocolatey and some hydra artifacts to run. This PR unifies the approach for Windows [by using MSYS2](https://github.com/msys2/setup-msys2).

Done in the same style as [this PR in `cardano-node`](https://github.com/input-output-hk/cardano-node/pull/4184)

# Checklist

- Branch
    - [X] Commit sequence broadly makes sense
    - [X] Commits have useful messages
    - [X] New tests are added if needed and existing tests are updated
    - [X] If this branch changes Consensus and has any consequences for downstream repositories or end users, said changes must be documented in [`interface-CHANGELOG.md`](../ouroboros-consensus/docs/interface-CHANGELOG.md)
    - [X] If this branch changes Network and has any consequences for downstream repositories or end users, said changes must be documented in [`interface-CHANGELOG.md`](../docs/interface-CHANGELOG.md)
    - [X] If serialization changes, user-facing consequences (e.g. replay from genesis) are confirmed to be intentional.
- Pull Request
    - [X] Self-reviewed the diff
    - [X] Useful pull request description at least containing the following information:
      - What does this PR change?
      - Why these changes were needed?
      - How does this affect downstream repositories and/or end-users?
      - Which ticket does this PR close (if any)? If it does, is it [linked](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)?
    - [X] Reviewer requested

### CODE REVIEW

**Review by amesgen — COMMENTED** [review]
Looks wonderful! 

Only other thing: Do you have evidence that caching is working? In [e4c66cb](https://github.com/input-output-hk/ouroboros-network/pull/3901/commits/e4c66cb4bd488429a5160e2dbef366a8b22e3405) (latest commit at time of writing), all dependencies are still built from scratch.

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ jobs:
 
     defaults:
       run:
-        shell: bash
+        shell: ${{ matrix.os == 'windows-latest' && 'msys2 {0}' || 'bash' }}
```
[the remark this answers is not in the document]
Comment (Jasagredo) [reply .github/workflows/build.yml]: Done! Nice suggestion, thanks!

**Review by amesgen — APPROVED** [review]
Hmm, unfortunate that `runner.os` doesn't work in the `shell:` expression :thinking: 

Either way, LGTM!

**Review by coot — APPROVED** [review]
LGTM

**Comment by coot** [comment]
@Jasagredo thanks for taking care of it.  Btw, what triggered moving away of `chocolatey`? It was a bit flaky at times, or was it something else?

**Comment by Jasagredo** [comment]
> @Jasagredo thanks for taking care of it. Btw, what triggered moving away of `chocolatey`? It was a bit flaky at times, or was it something else?

`chocolatey` was somewhat flaky, Also installing libsodium manually (sed replacing things) and also with libsecp256k1 felt more complex than doing the normal thing of `make install`. I think this new flow is much more uniform now

**Comment by Jasagredo** [comment]
bors retry

**Pull request merged**